spec/lita/handlers/github_prs/git_diff_spec.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+RSpec.describe Lita::Handlers::GithubPrs::GitDiff do
+  describe '#modified_files' do
+    it 'returns a list of modified files' do
+      files = [
+        build_file('modified_file.rb', status: 'modified'),
+        build_file('added_file.rb', status: 'added'),
+        build_file('deleted_file.rb', status: 'deleted')
+      ]
+      diff = build_diff(files: files)
+      git_diff = Lita::Handlers::GithubPrs::GitDiff.new(diff)
+
+      expect(git_diff.modified_files).to eq(['modified_file.rb'])
+    end
+  end
+
+  describe '#added_files' do
+    it 'returns a list of added files' do
+      files = [
+        build_file('modified_file.rb', status: 'modified'),
+        build_file('added_file.rb', status: 'added'),
+        build_file('deleted_file.rb', status: 'deleted')
+      ]
+      diff = build_diff(files: files)
+      git_diff = Lita::Handlers::GithubPrs::GitDiff.new(diff)
+
+      expect(git_diff.added_files).to eq(['added_file.rb'])
+    end
+  end
```

Comment:
I'm reflecting on this, not necessarily suggesting a change: if I were doing this I *might* have an outer `context` with the three files.  Again, it depends on the test file, but I think that part of the art of doing that is naming things well.  If you use the data (in this case the filename) as an opportunity to be descriptive about what that example is for (that is, having the file name be `added_file.rb` for the file that is added), there shouldn't be as much need to look up at the definitions.  Granted it may be important to understand that there are two other files that are included but their exclusion is part of what you're testing for.

Reply:
One thing that I like to do sometimes when pair programming is a particular style of [ping-pong pairing](https://tobeagile.com/2017/01/04/ping-pong-pairing/).  It's where one person writes tests and another person writes the code is an obstinately difficult way.  So, for example, to implement `added_files` I might implement it as `[@diff.files[1].filename]` to make this one test pass.  That forces the test writing pair to cover with more cases.  Of course you can get very ridiculous with that style, but the point is to push the tests somewhat to make them more thorough.

On the other hand for something that is relatively simple there isn't necessarily a lot of extra value in really thorough test coverage, so I play that by ear ;)

Reply:
I've been thinking about that style of using the data that is passed to the function within expectations and whether it is more explicit or not, it is tough to argue for one or the other as the arguments for both are good.

I've fallen into the group of preferring usage of raw values as much as possible with basically one simple argument: I won't have to run the test to be able to know what value is expected 🤷‍♀️